Sources/TreeSitterClient/TreeSitter+Extensions.swift

```diff
@@ -15,18 +15,25 @@ extension InputEdit {
             return nil
         }
 
-        guard
-            let startPoint = transformer(startLocation),
-            let newEndPoint = transformer(newEndLocation)
-        else {
-            return nil
+        let startPoint, newEndPoint: Point?
+        if let transformer = transformer {
+            startPoint = transformer(startLocation)
+            newEndPoint = transformer(newEndLocation)
+            if startPoint == nil || newEndPoint == nil {
+                return nil
+            }
+        } else {
+            startPoint = .zero
+            newEndPoint = .zero
         }
 
+        assert(startPoint != nil, "startPoint should not be nil")
+        assert(newEndPoint != nil, "newEndPoint should not be nil")
         self.init(startByte: UInt32(range.location * 2),
                   oldEndByte: UInt32(range.max * 2),
                   newEndByte: UInt32(newEndLocation * 2),
-                  startPoint: startPoint,
+                  startPoint: startPoint!,
                   oldEndPoint: oldEndPoint,
-                  newEndPoint: newEndPoint)
+                  newEndPoint: newEndPoint!)
```

Comment:
I think we can pull this off with a little less code. What do you think about this?

```swift
let startPoint = transformer?(startLocation)
let endPoint = transformer?(newEndLocation)

if transformer != nil {
    assert(startPoint != nil)
    assert(newEndPoint != nil)
}
```

and then in the initializer, we can use use optional chaining:

```
startPoint: startPoint ?? .zero
```

Reply:
Less code without losing brevity. Looks perfect.
I have updated the code